compiler/noirc_evaluator/src/acir/mod.rs

```diff
@@ -3691,4 +3691,203 @@ mod test {
 
         assert!(matches!(acvm.solve(), ACVMStatus::Failure::<FieldElement>(_)));
     }
+
+    // Convert the SSA input into ACIR and use ACVM to execute it
+    // Returns the ACVM execution status and the value of the 'output' witness value,
+    // unless the provided output is None or ACVM fail execution.
+    fn execute_ssa(
+        ssa: Ssa,
+        initial_witness: WitnessMap<FieldElement>,
+        output: Option<&Witness>,
+    ) -> (ACVMStatus<FieldElement>, Option<FieldElement>) {
+        let brillig = ssa.to_brillig(&BrilligOptions::default());
+        let (acir_functions, brillig_functions, _, _) = ssa
+            .into_acir(&brillig, &BrilligOptions::default(), ExpressionWidth::default())
+            .expect("Should compile manually written SSA into ACIR");
+        assert_eq!(acir_functions.len(), 1);
+        let main = &acir_functions[0];
+        let mut acvm = ACVM::new(
+            &StubbedBlackBoxSolver(true),
+            main.opcodes(),
+            initial_witness,
+            &brillig_functions,
+            &[],
+        );
+        let status = acvm.solve();
+        if status == ACVMStatus::Solved {
+            (status, output.map(|o| acvm.witness_map()[o]))
+        } else {
+            (status, None)
+        }
+    }
+
+    fn get_constant_main_src(v0: FieldElement, v1: FieldElement, typ: &str) -> String {
+        format!(
+            "acir(inline) fn main f0 {{
+            b0(v0: [{typ}; 2]):
```

Comment:
Can normal variable names be used here? E.g. `v0 -> input_array`, `v1 -> lhs`, etc.

Reply:
Done